### PR TITLE
JohnsonAlgorithm

### PR DESCRIPTION
**Pull request by saiadityaviswanadham**
Johnson Algorithm

## Related Issue
#148 

## Proposed Changes
Created and updated the johnsonalgorithm.cpp file
Uses existing Features Bellman Ford and Dijkstra's Algorithm to code Johnson Algorithm that can calculate the shortest path between two vertices. 

## Additional Info
- Johnson's Algorithm is the more effective for Sparse Graphs. It uses Bellman-Ford in order to reweight the input graph to eliminate negative edges and detect negative cycles. With this new, altered graph, it then uses Dijkstra's shortest path algorithm to calculate the shortest path between all pairs of vertices. The output of the algorithm is then the set of shortest paths in the original graph.

## Checklist

- [x] ✅ My code follows the code style of this project.
- [x] 📝 My change requires a change to the documentation.
- [x] 🎀 I have updated the documentation accordingly.
- [x] 👀 I have read the [**CONTRIBUTING**](https://github.com/Lakhankumawat/LearnCPP/blob/main/Contributing.md) document.
- [x] ✨ I have added tests to cover my changes.
- [x] 🚩 All new and existing tests passed.
- [x] 🌟 ed the repo

Created the johnsonalgorthm.cpp and framed the code as per the algorithm

This change is required as it helps to find shortest path between two vertices

Open Issue #148 

Using various Sample test cases and tracing them mathematically. 

Tested using GCC Compiler with the test cases.
Enter no of vertices: 3
Enter no of edges: 5
Enter the EDGE Costs:
1 2 8
2 1 12
1 3 22
3 1 6
2 3 4
Resultant adj matrix
0 8 12
10 0 4
6 14 0

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Output Screenshots
johnsonalgorithm

### CODE REVIEW

**Comment by saiadityaviswanadham** [comment]
This PR is with respect to the issue #148 .

**Comment by saiadityaviswanadham** [comment]
Updated the PR and added required details. Completed the checklist. Let me know if I missed anything..

**Comment by saiadityaviswanadham** [comment]
@Lakhankumawat Thankyou. I have completed it.

**Comment by Lakhankumawat** [comment]
> @Lakhankumawat Thankyou. I have completed it.

No You haven't Where is README ?????

**Comment by saiadityaviswanadham** [comment]
@Lakhankumawat Updated README.md in the GraphAlgorithms

File: G-GraphAlgorithms/S-SingleSourceShortestPath/johnsonalgorithm.cpp [hunk 1]
```diff
@@ -0,0 +1,38 @@
+#include<iostream>
+#include<bits/stdc++.h>
```
Comment (n4i9kita) [inline G-GraphAlgorithms/S-SingleSourceShortestPath/johnsonalgorithm.cpp:2]: Please use header files that are coming in use, and avoid using `bits/stdc++.h`.

**Review by n4i9kita — COMMENTED** [review]
LGTM

**Comment by saiadityaviswanadham** [comment]
@n4i9kita  @Lakhankumawat Updated Header Files

**Comment by aditi944** [comment]
closed

**Pull request closed without merging**